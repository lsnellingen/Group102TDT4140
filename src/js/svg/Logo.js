import React from 'react';

const Logo = () => (
  <svg width="40px" height="40px">
    <g>
    	<g>
    		<path id="XMLID_284_" fill="#71CAC3" d="M20.858,24.386h-4.755c-0.17,0-0.317-0.149-0.317-0.322V11.271
    			c0-0.173,0.146-0.321,0.317-0.321h4.755c0.168,0,0.314,0.149,0.314,0.321v12.793C21.172,24.237,21.026,24.386,20.858,24.386z"/>
    		<path fill="#333333" d="M38.165,28.637c0.144-0.387-0.032-0.813-0.414-0.971c-0.381-0.148-0.802,0.032-0.956,0.419
    			c-0.559,1.439-1.951,2.411-3.475,2.411H17.958c-0.186,0-0.373,0.073-0.511,0.203l-7.013,6.817v-6.282
    			c0-0.197-0.081-0.386-0.218-0.526c-0.137-0.14-0.323-0.222-0.518-0.222H6.2c-2.057,0-3.734-1.702-3.734-3.79V9.897
    			c0-2.088,1.676-3.791,3.734-3.791h16.643c0.405,0,0.734-0.337,0.734-0.748c0-0.412-0.33-0.749-0.734-0.749H6.2
    			C3.332,4.61,1,6.979,1,9.889v16.807c0,2.91,2.332,5.278,5.2,5.278h2.769v7.285c0,0.295,0.179,0.566,0.445,0.683
    			C9.503,39.983,9.609,40,9.706,40c0.187,0,0.374-0.074,0.511-0.206l8.042-7.819H33.33C35.443,31.983,37.388,30.634,38.165,28.637z"
    			/>
    		<path fill="#333333" d="M40.586,13.293c0.219-1.389-0.349-2.804-1.442-3.602c-0.623-0.451-1.352-0.69-2.112-0.69H31.66V5.103
    			c0-1.727-0.502-2.977-1.489-3.708c-1.564-1.167-3.702-0.501-3.792-0.477c-0.308,0.099-0.512,0.387-0.512,0.708v4.498
    			c0,3.462-3.974,4.646-4.136,4.695c-0.015,0.008-0.033,0.017-0.049,0.017c-0.243-0.378-0.663-0.634-1.149-0.634h-4.115
    			c-0.753,0-1.369,0.625-1.369,1.39v12.144c0,0.765,0.615,1.39,1.369,1.39h4.116c0.379,0,0.72-0.156,0.971-0.409
    			c0.656,0.663,1.564,1.075,2.567,1.075h10.837c2.453,0,4.018-1.307,4.301-3.585L40.586,13.293z M20.436,23.638h-3.912v-11.94h3.912
    			V23.638z M39.136,13.063l-1.385,8.929c0,0.011,0,0.016,0,0.024c-0.187,1.521-1.143,2.294-2.842,2.294H24.072
    			c-1.198,0-2.172-0.986-2.172-2.203v-9.826c0.075,0,0.147,0,0.222-0.024c0.209-0.058,5.214-1.531,5.214-6.135V2.234
    			c0.535-0.074,1.359-0.091,1.976,0.362c0.592,0.444,0.883,1.283,0.883,2.507V9.75c0,0.411,0.333,0.748,0.736,0.748h6.105
    			c0.456,0,0.894,0.14,1.257,0.412C38.935,11.378,39.266,12.224,39.136,13.063z"/>
          <text transform="matrix(1 0 0 1 42.9805 24.5137)" fill="#333333" fontFamily="'MyriadPro-Regular'" fontSize="21">EDUBOT</text>
    	</g>
    </g>
  </svg>
);

export default Logo;
