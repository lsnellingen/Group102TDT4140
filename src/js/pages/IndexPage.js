// Libs
import { Link } from 'react-router';
import React, { PropTypes } from 'react';
import DocumentTitle from 'react-document-title';
import { Authenticated, NotAuthenticated, LoginLink } from 'react-stormpath';
import axios from 'axios';

// Components
import Logo from '../svg/Logo';
import FeedbackIcon from 'react-icons/lib/md/feedback';
import QueriesIcon from 'react-icons/lib/fa/question';


export default class IndexPage extends React.Component {

  static contextTypes = {
    user: React.PropTypes.object
  };


  render() {
    const courses = this.context.user ? (this.context.user.customData.courses == undefined ? ["Du har ingen emner"] : this.context.user.customData.courses) : [];
    return (
      <div className="container">
        <div className="row">

          <div className="col-xs-6">
            <div className="jumbotron">
              <div className="row greyColor">
                <div>
                  <FeedbackIcon className="frontPageIcon" size={35}/>
                  <h3 className="inline">FEEDBACKS</h3>
                </div>

                <hr className="marginTop7" />
                <div>
                  <h5 className="frontPageSection"><big>As a students you can <strong>send</strong> specified feedback in all courses you are enrolled in. You can also <strong>view</strong> feedback
                    from other students. And if you agree? <strong>Like</strong> their feedback so they know they are not alone. If it get enough likes, maybe
                    your teacher will give you a <strong>response</strong>?
                  </big></h5>
                </div>
              </div>
            </div>
          </div>

<<<<<<< HEAD
          <Authenticated>
            <div className="col-xs-12">
              <div className="jumbotron">
                <div className="row">
                  <div className="col-xs-2 col-xs-offset-1">
                    <FeedbackIcon size={150} color={'#4C5760'}/>
                  </div>
                  <div className="col-xs-7 col-xs-offset-2">
                    <h2 className="text-center">EDUBOT - A feedback tool for students</h2>
                    <br />
                    <div className="vSpacing"><LabelIcon className="vCenterIcon" size={20} color={'#4C5760'}/><big>Create a user, register for courses and start sending feedback.</big></div>
                    <div className="vSpacing"><LabelIcon className="vCenterIcon" size={20} color={'#4C5760'}/><big>Choose a topic and tell your lecturer what you like and dislike.</big></div>
                    <div className="vSpacing"><LabelIcon className="vCenterIcon" size={20} color={'#4C5760'}/><big>Rate courses, other feedback and improve your own experience. </big></div>
                  </div>
=======
          <div className="col-xs-6">
            <div className="jumbotron">
              <div className="row greyColor">
                <div>
                  <QueriesIcon className="frontPageIcon" size={30}/>
                  <h3 className="inline">QUERIES</h3>
>>>>>>> ffed0909452e1101777a8a41c2784c5cf996193e
                </div>
                <hr className="marginTop7" />
                  <div>
                    <h5 className="frontPageSection"><big>As a lecturer you can <strong>make</strong> your own queries. <strong>Add</strong> as many questions as you like,
                      <strong> choose</strong> between different question types and <strong>ask</strong> your students. Wait and get <strong>responses</strong> from
                        your students on <strong>what you want to know</strong>.
                    </big></h5>
                  </div>
              </div>
            </div>
          </div>



          <Authenticated>

            <div className="container">
              <div className="col-xs-7">
                <div className="panel panel-info">
                  <div className="panel-heading">
                    <div className="">
                      <button className="btn btn-info pull-left">Recent feedback</button>

                      <button className="btn btn-info pull-right">Latest responses</button>
                      <button className="btn btn-info hCenter">Most upvoted</button>
                    </div>
                  </div>
                  <div className="panel-body">
                    <ul className="list-group">
                      <li className="list-group-item">First</li>
                      <li className="list-group-item">Second</li>
                      <li className="list-group-item">Third</li>
                    </ul>
                  </div>
                </div>
              </div>
              <div className="col-xs-5">
                <div className="panel panel-info">
                  <div className="panel-heading">
                    <h3 className="panel-title">User information:</h3>
                  </div>
                  <div className="panel-body">
                    <table className="table">
                      <tbody>
                        <tr>
                          <th>Name:</th>
                          <td>{this.context.user ? this.context.user.fullName : null}</td>
                        </tr>
                        <tr>
                          <th>Email:</th>
                          <td>{this.context.user ? this.context.user.email : null}</td>
                        </tr>
                        <tr>
                          <th>Position:</th>
                          <td>{this.context.user ? Object.keys(this.context.user.groups)[0] : null}</td>
                        </tr>
                        <tr>
                          <th>Member since:</th>
                          <td>{this.context.user ? this.context.user.createdAt.substring(0,10) : null}</td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>

          </Authenticated>

          <NotAuthenticated className="notAuthenticated">
            <div className="col-xs-7">
              <div className="panel panel-info">
                <div className="panel-heading">
                  <h4 className="text-center">Get started!</h4>
                </div>
                <div className="panel-body">
                  <h5>Start sending feedback to your lecturers, right away!</h5>
                  <p><em>
                    Create a user, sign up for courses and send feedback
                  </em></p>
                <Link to="/register"><button className="btn btn-default">REGISTER</button></Link>
                </div>
              </div>
            </div>
            <div className="col-xs-5">
              <div className="panel panel-info">
                <div className="panel-heading">
                  <h4 className="text-center">Already have a user?</h4>
                </div>
                <div className="panel-body">
                  <p>Let your lecturers know what you think, logg in and start sending feedback now.</p>
                  <Link to="/login"><button className="btn btn-default">LOGG INN</button></Link>
                </div>
              </div>
            </div>
          </NotAuthenticated>

        </div>
      </div>
    );
  }
}
