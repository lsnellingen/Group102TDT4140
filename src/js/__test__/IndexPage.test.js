// Libs
import { Link } from 'react-router';
import React, { PropTypes } from 'react';
import DocumentTitle from 'react-document-title';
import { Authenticated, NotAuthenticated, LoginLink } from 'react-stormpath';
import axios from 'axios';

// Components
import Logo from '../svg/Logo';
import FeedbackIcon from 'react-icons/lib/fa/comments';
import LabelIcon from 'react-icons/lib/md/label';

import IndexPage from '../pages/IndexPage';
import {shallow} from 'enzyme';
import ReactTestUtils from 'react-addons-test-utils';
import renderer from 'react-test-renderer';

let Example;
let onActionMock = jest.fn();
describe('IndexPage', () => {
  var ReactTestUtils = require('react-addons-test-utils') // ES5 with npm

  beforeEach(function() {
  onActionMock.mockClear();
  });
    test('renders without crashing', () => {
      /*const renderer = ReactTestUtils.createRenderer();*/

      const tree = renderer.create(
        <IndexPage/>).toJSON();


      expect(tree).toMatchSnapshot();




    });
    test('login-button links to appropriate page',() => {

      Example = renderer.create(
        <IndexPage/>
      );
      Example = ReactTestUtils.renderIntoDocument(<button/>);
      Example.doClick();
      /*ReactTestUtils.Simulate.click(<id{login-button"}/>);*/
    });
});
