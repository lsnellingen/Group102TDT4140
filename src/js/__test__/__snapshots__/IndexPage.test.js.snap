// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage renders without crashing 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="notAuthenticated"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <div
        className="col-xs-12"
      >
        <div
          className="jumbotron"
        >
          <div
            className="row"
          >
            <div
              className="col-xs-2 col-xs-offset-1"
            >
              <svg
                fill="currentColor"
                height={150}
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "color": "#4C5760",
                    "verticalAlign": "middle",
                  }
                }
                viewBox="0 0 40 40"
                width={150}
              >
                <g>
                  <path
                    d="m31.4 17.1q0 3.1-2.1 5.8t-5.7 4.1-7.9 1.6q-1.9 0-3.9-0.4-2.8 2-6.2 2.9-0.8 0.2-1.9 0.3h-0.1q-0.3 0-0.5-0.2t-0.2-0.4q0-0.1 0-0.2t0-0.1 0-0.1l0.1-0.2 0-0.1 0.1-0.1 0.1-0.1 0.1-0.1q0.1-0.1 0.5-0.6t0.6-0.6 0.5-0.7 0.6-0.8 0.4-1q-2.7-1.6-4.3-4t-1.6-5q0-3.1 2.1-5.7t5.7-4.2 7.9-1.5 7.9 1.5 5.7 4.2 2.1 5.7z m8.6 5.8q0 2.6-1.6 5t-4.3 3.9q0.2 0.5 0.4 1t0.6 0.8 0.5 0.7 0.6 0.7 0.5 0.5q0 0 0.1 0.1t0.1 0.1 0.1 0.1 0 0.2l0.1 0.1 0 0.1 0 0.1 0 0.2q0 0.3-0.3 0.5t-0.5 0.1q-1.1-0.1-1.9-0.3-3.4-0.9-6.2-2.9-2 0.4-3.9 0.4-6.1 0-10.6-3 1.3 0.1 2 0.1 3.6 0 6.9-1t5.9-2.9q2.8-2 4.3-4.7t1.5-5.7q0-1.7-0.5-3.4 2.9 1.6 4.5 4t1.7 5.2z"
                  />
                </g>
              </svg>
            </div>
            <div
              className="col-xs-7 col-xs-offset-2"
            >
              <h2
                className="text-center"
              >
                EDUBOT - A feedback tool for students
              </h2>
              <br />
              <div
                className="vSpacing"
              >
                <svg
                  className="vCenterIcon"
                  fill="currentColor"
                  height={20}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "color": "#4C5760",
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 40 40"
                  width={20}
                >
                  <g>
                    <path
                      d="m29.4 9.8l7.2 10.2-7.2 10.2c-0.6 0.9-1.7 1.4-2.8 1.4h-18.2c-1.8 0-3.4-1.4-3.4-3.2v-16.8c0-1.8 1.6-3.2 3.4-3.2h18.2c1.1 0 2.1 0.5 2.8 1.4z"
                    />
                  </g>
                </svg>
                <big>
                  Create a user, register for courses and start sending feedback.
                </big>
              </div>
              <div
                className="vSpacing"
              >
                <svg
                  className="vCenterIcon"
                  fill="currentColor"
                  height={20}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "color": "#4C5760",
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 40 40"
                  width={20}
                >
                  <g>
                    <path
                      d="m29.4 9.8l7.2 10.2-7.2 10.2c-0.6 0.9-1.7 1.4-2.8 1.4h-18.2c-1.8 0-3.4-1.4-3.4-3.2v-16.8c0-1.8 1.6-3.2 3.4-3.2h18.2c1.1 0 2.1 0.5 2.8 1.4z"
                    />
                  </g>
                </svg>
                <big>
                  Choose a subject and tell your lecturer what you like and dislike.
                </big>
              </div>
              <div
                className="vSpacing"
              >
                <svg
                  className="vCenterIcon"
                  fill="currentColor"
                  height={20}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "color": "#4C5760",
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 40 40"
                  width={20}
                >
                  <g>
                    <path
                      d="m29.4 9.8l7.2 10.2-7.2 10.2c-0.6 0.9-1.7 1.4-2.8 1.4h-18.2c-1.8 0-3.4-1.4-3.4-3.2v-16.8c0-1.8 1.6-3.2 3.4-3.2h18.2c1.1 0 2.1 0.5 2.8 1.4z"
                    />
                  </g>
                </svg>
                <big>
                  Rate courses, other feedback and improve your own experience. 
                </big>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-xs-7"
      >
        <div
          className="panel panel-info"
        >
          <div
            className="panel-heading"
          >
            <h4
              className="text-center"
            >
              Get started!
            </h4>
          </div>
          <div
            className="panel-body"
          >
            <h5>
              Start sending feedback to your lecturers, right away!
            </h5>
            <p>
              <em>
                Create a user, sign up for courses and send feedback
              </em>
            </p>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <button
                className="btn btn-default"
              >
                REGISTER
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        className="col-xs-5"
      >
        <div
          className="panel panel-info"
        >
          <div
            className="panel-heading"
          >
            <h4
              className="text-center"
            >
              Already have a user?
            </h4>
          </div>
          <div
            className="panel-body"
          >
            <p>
              Let your lecturers know what you think, logg in and start sending feedback now.
            </p>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <button
                className="btn btn-default"
              >
                LOGG INN
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
